tutorial_code/example3_2_tafm_forward_anneal.py

```diff
@@ -318,132 +362,204 @@ def run_experiment(param, shim, stats, embeddings, logical_bqm, alpha_Phi=0., al
+                    shim["alpha_J"] = alpha_J
 
                 run_iteration(param, shim, stats, embeddings, logical_bqm)
 
         else:
             # Adaptive step sizes
-
-            shim['alpha_Phi'] = alpha_Phi
-            shim['alpha_J'] = alpha_J
+            shim["alpha_Phi"] = alpha_Phi
+            shim["alpha_J"] = alpha_J
             for iteration in pbar:
                 run_iteration(param, shim, stats, embeddings, logical_bqm)
-                shim['alpha_Phi'] *= shim_parameter_rescaling(stats['all_fbos'], num_iters=20,
-                                                              ratio=1.1)
-                shim['alpha_J'] *= shim_parameter_rescaling(stats['all_couplings'], num_iters=20,
-                                                            ratio=1.1)
-
-        save_experiment_data(
-            prefix,
-            {'param': param, 'shim': shim, 'stats': stats}
-        )
+                shim["alpha_Phi"] *= shim_parameter_rescaling(
+                    stats["all_fbos"], num_iters=20, ratio=1.1
+                )
+                shim["alpha_J"] *= shim_parameter_rescaling(
+                    stats["all_couplings"], num_iters=20, ratio=1.1
+                )
+        if use_cache:
+            save_experiment_data(prefix, {"param": param, "shim": shim, "stats": stats})
+
+    paper_plots_example3_2(
+        halve_boundary_couplers=param["halve_boundary_couplers"],
+        type_=shim["type"],
+        nominal_couplings=shim["nominal_couplings"],
+        coupler_orbits=shim["coupler_orbits"],
+        all_fbos=stats["all_fbos"],
+        all_couplings=stats["all_couplings"],
+        mags=stats["mags"],
+        frust=stats["frust"],
+    )
 
-    plot_data(all_fbos=stats['all_fbos'], mags=stats['mags'],
-              all_couplings=stats['all_couplings'], frust=stats['frust'],
-              all_alpha_phi=stats['all_alpha_Phi'], all_alpha_j=stats["all_alpha_J"],
-              coupler_orbits=shim['coupler_orbits'], alpha_phi=shim['alpha_Phi'], alpha_j=shim['alpha_J'],
-              coupling=param["coupling"], L=param["L"])
-    paper_plots_example3_2(halve_boundary_couplers=param['halve_boundary_couplers'],
-                           type_=shim['type'], nominal_couplings=shim['nominal_couplings'],
-                           coupler_orbits=shim['coupler_orbits'], all_fbos=stats['all_fbos'],
-                           all_couplings=stats['all_couplings'], mags=stats['mags'],
-                           frust=stats['frust'], all_psi=stats['all_psi'])
+    return {
+        "halve_boundary_couplers": param["halve_boundary_couplers"],
+        "type_": shim["type"],
+        "all_psi": stats["all_psi"],
+        "shim_type": shim["type"],
+    }
 
 
-def main():
-    """Main function to run example
+def main(
+    solver_name: str = None,
+    coupling: float = 0.9,
+    num_iters: int = 800,
+    num_iters_unshimmed_flux: int = 100,
+    num_iters_unshimmed_J: int = 300,
+    max_num_emb: int = 1,
```

Comment:
This is how I would like to see max_num_emb exposed in the other example*.py files. but for embed_loops we default to float('Inf')

Reply:
`embed_loops` and `embed_sqaure_lattice` now has max_num_emb in main function defaulted to float('inf'). All other example scripts has max_num_emb defaulted to 1.